tests/drivers/build_all/sensor/src/generic_test.c

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/emul_sensor.h>
+#include <zephyr/drivers/emul.h>
+#include <zephyr/ztest.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/rtio/rtio.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(generic_test);
+
+/*
+ * Set up an RTIO context that can be shared for all sensors
+ */
+
+static enum sensor_channel iodev_all_channels[SENSOR_CHAN_ALL];
+static struct sensor_read_config iodev_read_config = {
+	.channels = iodev_all_channels,
+	.max = SENSOR_CHAN_ALL,
+};
+RTIO_IODEV_DEFINE(iodev_read, &__sensor_iodev_api, &iodev_read_config);
+
+/* Create the RTIO context to service the reading */
+RTIO_DEFINE_WITH_MEMPOOL(sensor_read_rtio_ctx, 1, 1, 1, 64, 4);
+
+/** Number of test points to use in the generic test. These are automatically generated. */
+#define NUM_EXPECTED_VALS (5)
+
+/**
+ * @brief Prepare the RTIO context for next test
+ */
+static void before(void *args)
```

Comment:
If a test fails, the mempool associated with the RTIO context will not be reset, you should clear all the bits in the associated bitmap here.

Also, flush out any pending SQE's/ CQE's